### ISSUE TITLE
Fedora 26 Kernel  > 4.11.11-200.fc26.x86_64 breaks Bumblebee, how to debug?

### ISSUE THREAD

**Issue by aryasenna** [Bumblebee-Project/Bumblebee#905]
So upon kernel update from  kernel-4.11.11-200 to kernel-4.11.11-300
I noticed Bumblebeed failed to start:

> Aug 11 19:36:02 localhost systemd[1]: bumblebee-nvidia.service: Failed with result 'exit-code'.
> Aug 11 19:36:02 localhost systemd[1]: bumblebee-nvidia.service: Unit entered failed state.
> Aug 11 19:36:02 localhost systemd[1]: Failed to start Compiling NVidia Driver.
> Aug 11 19:36:02 localhost audit[1]: SERVICE_START pid=1 uid=0 auid=4294967295 ses=4294967295 subj=system_u:system_r:init_t:s0 msg='unit=bumblebee-nvidia 
> Aug 11 19:36:02 localhost systemd[1]: bumblebee-nvidia.service: Main process exited, code=exited, status=1/FAILURE
> Aug 11 19:36:02 localhost bumblebee-nvidia[2492]: 

I then run `bumblebee --debug`

> --debug mode selected.
> Building NVIDIA video drivers: Creating directory NVIDIA-Linux-x86_64-384.59
> Verifying archive integrity... OK
> Uncompressing NVIDIA Accelerated Graphics Driver for Linux-x86_64 384.59..........................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................
> 
> ERROR: Unable to load the kernel module 'nvidia.ko'.  This happens most
>        frequently when this kernel module was built against the wrong or
>        improperly configured kernel sources, with a version of gcc that differs
>        from the one used to build the target kernel, or if a driver such as
>        rivafb, nvidiafb, or nouveau is present and prevents the NVIDIA kernel
>        module from obtaining ownership of the NVIDIA graphics device(s), or no
>        NVIDIA GPU installed in this system is supported by this NVIDIA Linux
>        graphics driver release.
>        
>        Please see the log entries 'Kernel module load error' and 'Kernel
>        messages' at the end of the file '/var/log/nvidia-installer.log' for
>        more information.
> 
> 
> ERROR: Installation has failed.  Please see the file
>        '/var/log/nvidia-installer.log' for details.  You may find suggestions
>        on fixing installation problems in the README available on the Linux
>        driver download page at www.nvidia.com.

The kernel log:

> Aug 11 19:37:31 localhost kernel: nvidia-nvlink: Unregistered the Nvlink Core, major device number 238
> 
> Aug 11 19:37:31 localhost kernel: NVRM: None of the NVIDIA graphics adapters were initialized!
> Aug 11 19:37:31 localhost kernel: NVRM: The NVIDIA probe routine failed for 1 device(s).
> Aug 11 19:37:31 localhost kernel: nvidia: probe of 0000:01:00.0 failed with error -1
> Aug 11 19:37:31 localhost kernel: NVRM: The NVIDIA GPU 0000:01:00.0
>                                NVRM: (PCI ID: 10de:0ff6) installed in this system has
>                                NVRM: fallen off the bus and is not responding to commands.
> Aug 11 19:37:31 localhost kernel: nvidia 0000:01:00.0: vgaarb: changed VGA decodes: olddecodes=io+mem,decodes=none:owns=none
> Aug 11 19:37:31 localhost kernel: nvidia 0000:01:00.0: enabling device (0000 -> 0003)
> Aug 11 19:37:31 localhost kernel: nvidia-nvlink: Nvlink Core is being initialized, major device number 238
> Aug 11 19:37:31 localhost kernel: Disabling lock debugging due to kernel taint
> Aug 11 19:37:31 localhost kernel: nvidia: module license 'NVIDIA' taints kernel.
> Aug 11 19:37:31 localhost kernel: VFIO - User Level meta-driver version: 0.3

When I booted using the old kernel  (4.11.11-200), then do another `--debug` the nvidia.ko module loads fine and bumblebee is working again.

Any idea how to troubleshoot this? is this caused by changes in fedora kernel, mainline kernel, or it's just my unfortunate configuration?

The latest kernel version offered by default F26 repo is kernel-4.11.12-200.fc25.x86_64, I am facing same issue with the 'fallen off the bus ' error.

My video card is NVIDIA Quadro K1200M, I installed bumblebee from the repo mentioned here: https://fedoraproject.org/wiki/Bumblebee

My version of bumblebee:

>  bumblebee-3.2.1-13.fc26.x86_64
> bumblebee-release-1.2-1.noarch
> bumblebee-nvidia-384.59-1.fc26.x86_64
> bumblebee-nonfree-release-1.2-1.noarch

Please let me know if you need more info.

**Comment by aryasenna** [comment]
I believe the Fedora bumblebee is packaged by @gsgatlin. Perhaps he can shed some light?

**Comment by gsgatlin** [comment]
Hello. I am using kernel 4.11.11-300.fc26.x86_64

Can you check and see if kernel-devel-4.11.11-300.fc26.x86_64 is installed?

can you post /var/log/nvidia-installer.log

to pastebin.com or https://paste.fedoraproject.org/ or similar site?

Thanks.

**Comment by aryasenna** [comment]
Hi,

Looks like it's only on my setup, then?

Yes, kernel-devel is installed

```
user ~ rpm -qa kernel-devel    
kernel-devel-4.11.11-300.fc26.x86_64
kernel-devel-4.11.12-200.fc25.x86_64
kernel-devel-4.11.11-200.fc25.x86_64
```

nvidia-installer.log  --> https://paste.fedoraproject.org/paste/PO3ZjqtuO9RuXNthUiYCWA.
AFAICT the kernel module compiles just fine,  but it fails when it tries to modprobe.

**Comment by gsgatlin** [comment]
Oh look: in the file it has

kernel source path: '/lib/modules/4.11.12-200.fc25.x86_64/source'
-> Kernel output path: '/lib/modules/4.11.12-200.fc25.x86_64/build'

I wonder if something got messed up in your kernel upgrade? To make it think it should use that?

Like I would think that should be

kernel source path: '/lib/modules/4.11.12-300.fc25.x86_64/source'
-> Kernel output path: '/lib/modules/4.11.12-300.fc25.x86_64/build'

I have a fedora 25 laptop at work. I'll try to upgrade the kernel on that and check to see if that works for me. I'm going over there tonight so I'll grab that laptop.

**Comment by gsgatlin** [comment]
Like I have a single

/usr/src/kernels/4.11.11-300.fc26.x86_64/

under

/usr/src/kernels/

which is what I'm guessing it uses.

**Comment by aryasenna** [comment]
Ah that log was made when I was booted to F26 using kernel-4.11.12-200.fc25.x86_64.
So I think it was the right kernel source being build.

Here right now I'm using the latest kernel 4.11.11-300.fc26.x86_64
Same problem: https://paste.fedoraproject.org/paste/5LvrVsXsf6-ECRZ~AkyQiA

The point is, for my setup bumblebee only works when I boot to F26 using 4.11.11-200.
Something like this

4.11.11-300.fc26.x86_64 -> doesn't work
4.11.12-200.fc25.x86_64 -> doesn't work
4.11.11-200.fc25.x86_64 -> works!

**Comment by gsgatlin** [comment]
How about the output from

bumblebee-nvidia --check

and

rpm -qa | grep kernel

if you are still on fedora 25 I'd guess we would want to use 4.11.12-200.fc25.x86_64 and remove 4.11.11-300.fc26.x86_64 when doing the tests and rpm command.

**Comment by gsgatlin** [comment]
oh and if you upgraded to fedora 26 then use 4.11.11-300.fc26.x86_64 . cheers.

**Comment by aryasenna** [comment]
I am using F26 with 4.11.11-300.fc26.x86_64  now.

```
aryasenna ~  sudo bumblebee-nvidia --check
--force compile selected via /etc/sysconfig/nvidia/compile-nvidia-driver
Warning! This NVIDIA driver has not compiled successfully before on kernel 4.11.11-300.fc26.x86_64!
Warning! This NVIDIA driver userland
 /usr/lib64/nvidia-bumblebee/libGL.so.1 library is missing!


Warning! nvidia.ko was not created in in the kernel tree!
modinfo output for NVIDIA:

filename:       /lib/modules/4.11.11-300.fc26.x86_64/kernel/drivers/video/nvidia.ko
modinfo: ERROR: could not get modinfo from 'nvidia': No such file or directory

Check bbswitch kernel module...

bbswitch is loaded into the current kernel ok.

All NVIDIA checks completed, but there were 1 or more failures...
Try running this script with the --debug option to find clues about what has
gone wrong with the NVIDIA driver compile process.

```
```
 aryasenna  ~  rpm -qa | grep kernel
kernel-4.11.11-200.fc25.x86_64
kernel-modules-extra-4.11.11-300.fc26.x86_64
kernel-modules-extra-4.11.11-200.fc25.x86_64
kernel-modules-extra-4.11.12-200.fc25.x86_64
kernel-devel-4.11.11-300.fc26.x86_64
kernel-4.11.11-300.fc26.x86_64
kernel-modules-4.11.11-300.fc26.x86_64
abrt-addon-kerneloops-2.10.3-1.fc26.x86_64
texlive-l3kernel-svn41246-33.fc26.2.noarch
kernel-core-4.11.11-200.fc25.x86_64
libreport-plugin-kerneloops-2.9.1-2.fc26.x86_64
kernel-modules-4.11.12-200.fc25.x86_64
kernel-4.11.12-200.fc25.x86_64
kernel-modules-4.11.11-200.fc25.x86_64
kernel-devel-4.11.12-200.fc25.x86_64
kernel-headers-4.11.11-300.fc26.x86_64
kernel-core-4.11.12-200.fc25.x86_64
kernel-core-4.11.11-300.fc26.x86_64

```

**Comment by gsgatlin** [comment]
Oh ok. Hmnn. It looks to me like you have all the rpm packages so that it should work ok. So its not that I don't think. You have:

kernel-devel-4.11.11-300.fc26.x86_64
kernel-4.11.11-300.fc26.x86_64
kernel-modules-4.11.11-300.fc26.x86_64
kernel-headers-4.11.11-300.fc26.x86_64
kernel-core-4.11.11-300.fc26.x86_64

I have these on my fedora 26 laptop also.

[gsgatlin@y470b ~]$ rpm -qa | grep kernel | grep 4.11.11-300
kernel-4.11.11-300.fc26.x86_64
kernel-core-4.11.11-300.fc26.x86_64
kernel-headers-4.11.11-300.fc26.x86_64
kernel-tools-libs-4.11.11-300.fc26.x86_64
kernel-modules-extra-4.11.11-300.fc26.x86_64
kernel-devel-4.11.11-300.fc26.x86_64
kernel-modules-4.11.11-300.fc26.x86_64

In the past when I have encountered a a problem where the module compiles but does not load, a reboot has always fixed this for me. But I assume it did not help you out...

Can you run?

journalctl

as root and then run

 bumblebee-nvidia --debug

also as root.

Perhaps there could be some output that would give clues as to why it will not load into the current kernel  in the journalctl. Can you then paste the output from that journalctl run to pastebin.com or https://paste.fedoraproject.org/ or similar site?

Also, secure boot should be off in the bios but I'm sure you did that already if it was working before. Just wanted to cover all bases since that can be another reason the module won't load.

Thanks.

**Comment by aryasenna** [comment]
So I just removed all bumblebee\* primus, virtuagl\*, reboot then install them again. Same log, same error message, no go.

Let see, I would address your question them one by one, that's a good idea to cover all bases:

>  reboot has always fixed this for me. But I assume it did not help you out...
Just rebooted (again), no go

> journalctl as root

`[root@localhost aryasenna] # journalctl`
https://paste.fedoraproject.org/paste/jdF7dQIiIAAxVtC3gbYa7Q

>bumblebee-nvidia --debug also as root.

`[root@localhost aryasenna] #  bumblebee-nvidia --debug`
https://paste.fedoraproject.org/paste/2qzRQgjRrtoOohwvYqEP4g

Just in case you need it again:
`[root@localhost aryasenna] # cat /var/log/nvidia-installer.log `
https://paste.fedoraproject.org/paste/~2l2wGy6fvNAB0v416Nr7g

> secure boot should be off in the bios 

Yes it is off. Checked the BIOS. 

Basically the logs are identical with the ones I have posted in previous reply except the timestamp.

```
[ 4428.586145] nvidia 0000:01:00.0: vgaarb: changed VGA decodes: olddecodes=none,decodes=none:owns=none
[ 4428.586175] NVRM: The NVIDIA GPU 0000:01:00.0
               NVRM: (PCI ID: 10de:0ff6) installed in this system has
               NVRM: fallen off the bus and is not responding to commands.
``

These errors bugs me, Google says it's due to hardware defect, but it runs fine with Games and  graphic intensive program in Windows, also when I boot to kernel-4.11.11-200.fc25.x86_64, the modprobe always succeeded. Hmmm...

**Comment by aryasenna** [comment]
@gsgatlin  I see the 'nvidia' is blacklisted on /etc/modprobe.d
Is this normal? 

```
[root@localhost aryasenna] # cat /etc/modprobe.d/bumblebee.conf 
blacklist nvidia
blacklist nouveau

```

**Comment by gsgatlin** [comment]
Yeah. That is normal. The developers told me to add the blacklist for both modules. Here it is on my machine

[gsgatlin@y470b ~]$ cat /etc/modprobe.d/bumblebee.conf 
blacklist nvidia
blacklist nouveau

https://github.com/Bumblebee-Project/Bumblebee/issues/582#issuecomment-70429951

**Comment by arcivanov** [comment]
Cross-referencing: https://devtalk.nvidia.com/default/topic/1022369/linux/dell-precision-7510-384-59-fedora-26-v-4-11-11-300-optimus-bumblebee-fallen-off-the-bus

**Comment by arcivanov** [comment]
@gsgatlin could be related to #172?

**Comment by gsgatlin** [comment]
Sorry. Did not see the log from journalctl. Also sorry for the delay in replying. Can you run

journalctl

and at the same time run bumblebee-nvidia and then paste the output from journalctl from the time you ran bumblebee-nvidia until it fails to a site such as pastebin.com or https://paste.fedoraproject.org/ or similar site?

The  link labled journalctl above actually is a link to the output from /var/log/nvidia-installer.log I think.

Does adding

pcie_port_pm=off

help as referenced by @arcivanov  at the devtalk.nvidia.com site?

**Comment by aryasenna** [comment]
@arcivanov Aha, another Quadro card, could it be this issue is isolated to Quadro cards? I don't see people with GeForce complaining, in fact @gsgatlin's Lenovo Y-series is running Bumblebee just fine with kernel-4.11.11-300.fc26.x86_64.

@gsgatlin 
No, the output of journalctl is exactly that: large part of it is coming from nvidia-installer.log. (cue the tail command, but  I do NOT manually type that.) it is journalct output _as is_.

Ah yes, I forget to tell you I have tried adding `pcie_port_pm=off` in kernel parameter. Even manually echoing 'on' to "" just to be sure this has nothing to do with PCI runtime power management.

Either attempt makes no difference, (also on the Fedora Bumblebee wiki it is said the problem related to pcie_port_pm=off has been fixed on kernel 4.8).

I'm inclined to say, this has something to do with changes in kernel, since nVidia driver/Bumblebee keeps working smoothly whenever I boot to kernel-4.11.11-200.fc25.x86_64. Change it to kernel-4.12.11-200.fc25.x86_64 or 4.11.11-300.fc26.x86_64 , bam "fallen of the bus".

Maybe others with non Fedora kernel 4.11.11 + Quadro + Bumblebee could confirm this.

**Comment by arcivanov** [comment]
@gsgatlin

```
[   41.770230] VFIO - User Level meta-driver version: 0.3
[   41.783195] nvidia: module license 'NVIDIA' taints kernel.
[   41.783197] Disabling lock debugging due to kernel taint
[   41.790717] nvidia-nvlink: Nvlink Core is being initialized, major device number 236
[   41.790942] nvidia 0000:01:00.0: vgaarb: changed VGA decodes: olddecodes=io+mem,decodes=none:owns=none
[   41.791022] NVRM: loading NVIDIA UNIX x86_64 Kernel Module  384.59  Wed Jul 19 23:53:34 PDT 2017 (using threaded interrupts)
[   41.801975] nvidia-uvm: Loaded the UVM driver in 8 mode, major device number 235
[   41.804071] nvidia-modeset: Loading NVIDIA Kernel Mode Setting Driver for UNIX platforms  384.59  Wed Jul 19 23:46:42 PDT 2017
[   41.807648] [drm] [nvidia-drm] [GPU ID 0x00000100] Loading driver
[   41.807650] [drm] Initialized nvidia-drm 0.0.0 20160202 for 0000:01:00.0 on minor 1
[   41.810505] [drm] [nvidia-drm] [GPU ID 0x00000100] Unloading driver
[   41.846363] nvidia-modeset: Unloading
[   41.863872] nvidia-uvm: Unloaded the UVM driver in 8 mode
[   41.881347] nvidia-nvlink: Unregistered the Nvlink Core, major device number 236
[   48.526932] nvidia-nvlink: Nvlink Core is being initialized, major device number 236
[   48.527193] nvidia 0000:01:00.0: vgaarb: changed VGA decodes: olddecodes=none,decodes=none:owns=none
[   48.527303] NVRM: loading NVIDIA UNIX x86_64 Kernel Module  384.59  Wed Jul 19 23:53:34 PDT 2017 (using threaded interrupts)
[   48.528679] nvidia-modeset: Loading NVIDIA Kernel Mode Setting Driver for UNIX platforms  384.59  Wed Jul 19 23:46:42 PDT 2017
[   48.531404] [drm] [nvidia-drm] [GPU ID 0x00000100] Loading driver
[   48.531406] [drm] Initialized nvidia-drm 0.0.0 20160202 for 0000:01:00.0 on minor 1
[   48.667997] [drm] [nvidia-drm] [GPU ID 0x00000100] Unloading driver
[   48.707118] nvidia-modeset: Unloading
[   48.731003] nvidia-nvlink: Unregistered the Nvlink Core, major device number 236
[   48.754811] bbswitch: disabling discrete graphics
[   48.754821] ACPI Warning: \_SB.PCI0.PEG0.PEGP._DSM: Argument #4 type mismatch - Found [Buffer], ACPI requires [Package] (20170303/nsarguments-95)
[   48.766844] pci_raw_set_power_state: 2 callbacks suppressed
[   48.766846] pci 0000:01:00.0: Refused to change power state, currently in D0
[  110.068087] bbswitch: enabling discrete graphics
[  110.420641] nvidia-nvlink: Nvlink Core is being initialized, major device number 236
[  110.420847] nvidia 0000:01:00.0: vgaarb: changed VGA decodes: olddecodes=none,decodes=none:owns=none
[  110.420926] NVRM: loading NVIDIA UNIX x86_64 Kernel Module  384.59  Wed Jul 19 23:53:34 PDT 2017 (using threaded interrupts)
[  110.444476] ACPI Warning: \_SB.PCI0.PEG0.PEGP._DSM: Argument #4 type mismatch - Found [Buffer], ACPI requires [Package] (20170303/nsarguments-95)
[  110.444537] ACPI Warning: \_SB.PCI0.PEG0.PEGP._DSM: Argument #4 type mismatch - Found [Buffer], ACPI requires [Package] (20170303/nsarguments-95)
[  110.444577] ACPI Warning: \_SB.PCI0.PEG0.PEGP._DSM: Argument #4 type mismatch - Found [Buffer], ACPI requires [Package] (20170303/nsarguments-95)
[  110.444702] ACPI Warning: \_SB.PCI0.PEG0.PEGP._DSM: Argument #4 type mismatch - Found [Buffer], ACPI requires [Package] (20170303/nsarguments-95)
[  110.444738] ACPI Warning: \_SB.PCI0.PEG0.PEGP._DSM: Argument #4 type mismatch - Found [Buffer], ACPI requires [Package] (20170303/nsarguments-95)
[  110.444784] ACPI Warning: \_SB.PCI0.PEG0.PEGP._DSM: Argument #4 type mismatch - Found [Buffer], ACPI requires [Package] (20170303/nsarguments-95)
[  110.444821] ACPI Warning: \_SB.PCI0.PEG0.PEGP._DSM: Argument #4 type mismatch - Found [Buffer], ACPI requires [Package] (20170303/nsarguments-95)
[  111.329735] ACPI Warning: \_SB.PCI0.PEG0.PEGP._DSM: Argument #4 type mismatch - Found [Buffer], ACPI requires [Package] (20170303/nsarguments-95)
[  111.329978] ACPI Warning: \_SB.PCI0.PEG0.PEGP._DSM: Argument #4 type mismatch - Found [Buffer], ACPI requires [Package] (20170303/nsarguments-95)
[  111.345069] ACPI Warning: \_SB.PCI0.PEG0.PEGP._DSM: Argument #4 type mismatch - Found [Buffer], ACPI requires [Package] (20170303/nsarguments-95)
[  111.745662] nvidia-modeset: Loading NVIDIA Kernel Mode Setting Driver for UNIX platforms  384.59  Wed Jul 19 23:46:42 PDT 2017
[  131.914080] ACPI Warning: \_SB.PCI0.PEG0.PEGP._DSM: Argument #4 type mismatch - Found [Buffer], ACPI requires [Package] (20170303/nsarguments-95)
[  131.920250] nvidia-modeset: Unloading
[  131.938087] nvidia-nvlink: Unregistered the Nvlink Core, major device number 236
[  131.958784] bbswitch: disabling discrete graphics
[  131.958792] ACPI Warning: \_SB.PCI0.PEG0.PEGP._DSM: Argument #4 type mismatch - Found [Buffer], ACPI requires [Package] (20170303/nsarguments-95)
[  131.970801] pci 0000:01:00.0: Refused to change power state, currently in D0
```
`pcie_port_pm=off` helps.

**Comment by aryasenna** [comment]
> pcie_port_pm=off helps. 

yes, it works, just tested it.

``

**Comment by gsgatlin** [comment]
Oh. Ok. awesome. Yeah. sorry this stuff is so freaking weird/complex.

**Comment by aryasenna** [comment]
Well should I close this issue since the workarounfd has been found and this is related to kernel change?

@gsgatlin yeah it's working again at last. :)

**Comment by gsgatlin** [comment]
Ok. sure. I will try to update the wiki so it doesn't mention 4.8. But yeah. Feel free to close this issue.

**Comment by aryasenna** [comment]
Reopening the issue @gsgatlin @arcivanov 
so I started the boot _**without**_ 'pcie_port_pm=off 

and Bumblebee runs fine.

``` 
aryasenna  ~ dmesg | grep "Command line"               
[    0.000000] Command line: BOOT_IMAGE=/vmlinuz-4.11.11-300.fc26.x86_64 root=/dev/mapper/fedora-root ro rd.lvm.lv=fedora/root rd.lvm.lv=fedora/swap nouveau.modeset=0 rd.driver.blacklist=nouveau rhgb quiet i915.i915_enable_rc6=1 i915.i915_enable_fbc=1 drm.vblankoffdelay=1 resume=/dev/dm-1 nouveau.blacklist=1
```
```
aryasenna  ~  uname -r
4.11.11-300.fc26.x86_64
```

```
 aryasenna ~  sudo bumblebee-nvidia --check
[sudo] password for aryasenna: 


nvidia.ko compiled into in the kernel tree ok.
modinfo output for NVIDIA:

filename:       /lib/modules/4.11.11-300.fc26.x86_64/kernel/drivers/video/nvidia.ko
alias:          char-major-195-*
version:        384.59
supported:      external
license:        NVIDIA
srcversion:     31FF0349D3C7B1D9A62B474
alias:          pci:v000010DEd00000E00sv*sd*bc04sc80i00*
alias:          pci:v000010DEd*sv*sd*bc03sc02i00*
alias:          pci:v000010DEd*sv*sd*bc03sc00i00*
depends:        
vermagic:       4.11.11-300.fc26.x86_64 SMP mod_unload 
parm:           NVreg_Mobile:int
parm:           NVreg_ResmanDebugLevel:int
parm:           NVreg_RmLogonRC:int
parm:           NVreg_ModifyDeviceFiles:int
parm:           NVreg_DeviceFileUID:int
parm:           NVreg_DeviceFileGID:int
parm:           NVreg_DeviceFileMode:int
parm:           NVreg_UpdateMemoryTypes:int
parm:           NVreg_InitializeSystemMemoryAllocations:int
parm:           NVreg_UsePageAttributeTable:int
parm:           NVreg_MapRegistersEarly:int
parm:           NVreg_RegisterForACPIEvents:int
parm:           NVreg_CheckPCIConfigSpace:int
parm:           NVreg_EnablePCIeGen3:int
parm:           NVreg_EnableMSI:int
parm:           NVreg_TCEBypassMode:int
parm:           NVreg_UseThreadedInterrupts:int
parm:           NVreg_EnableStreamMemOPs:int
parm:           NVreg_MemoryPoolSize:int
parm:           NVreg_RegistryDwords:charp
parm:           NVreg_RegistryDwordsPerDevice:charp
parm:           NVreg_RmMsg:charp
parm:           NVreg_AssignGpus:charp

Check bbswitch kernel module...

bbswitch is loaded into the current kernel ok.

All checks completed successfully! NVIDIA driver appears to have compiled ok.

Documentation on bumblebee for RHEL / CentOS / fedora can be found at:
https://www.linux.ncsu.edu/bumblebee/
```

```
aryasenna  ~  optirun glxinfo | grep NVIDIA
OpenGL vendor string: NVIDIA Corporation
OpenGL core profile version string: 4.5.0 NVIDIA 384.59
OpenGL core profile shading language version string: 4.50 NVIDIA
OpenGL version string: 4.5.0 NVIDIA 384.59
OpenGL shading language version string: 4.50 NVIDIA
```

So that `pcie_port_pm=off ` trick is either just coincidence/just related, or it is only needed for initial compilation.
In any case Bumblebee is running without that flag in my setup.

**Comment by arcivanov** [comment]
@aryasenna I'm not on 11. I'm on 4.12.5-300.fc26.

**Comment by arcivanov** [comment]
@aryasenna 

```
bash-4.4$ optirun glxinfo
[  170.504560] [ERROR]Cannot access secondary GPU - error: Could not load GPU driver

[  170.504597] [ERROR]Aborting because fallback start is disabled.
```
```
[  169.276541] bbswitch: enabling discrete graphics
[  170.197828] nvidia-nvlink: Nvlink Core is being initialized, major device number 238
[  170.198084] nvidia 0000:01:00.0: vgaarb: changed VGA decodes: olddecodes=none,decodes=none:owns=none
[  170.198108] NVRM: The NVIDIA GPU 0000:01:00.0
               NVRM: (PCI ID: 10de:13b0) installed in this system has
               NVRM: fallen off the bus and is not responding to commands.
[  170.198115] nvidia: probe of 0000:01:00.0 failed with error -1
[  170.198133] NVRM: The NVIDIA probe routine failed for 1 device(s).
[  170.198134] NVRM: None of the NVIDIA graphics adapters were initialized!
[  170.198233] nvidia-nvlink: Unregistered the Nvlink Core, major device number 238
```

**Comment by aryasenna** [comment]
Just upgraded to 4.12.5-300.fc26, except mine _is_ working.

Still without `pcie_pm_port`
```
aryasenna  ~ cat /proc/cmdline
BOOT_IMAGE=/vmlinuz-4.12.5-300.fc26.x86_64 root=/dev/mapper/fedora-root ro rd.lvm.lv=fedora/root rd.lvm.lv=fedora/swap nouveau.modeset=0 rd.driver.blacklist=nouveau rhgb quiet i915.i915_enable_rc6=1 i915.i915_enable_fbc=1 drm.vblankoffdelay=1 resume=/dev/dm-1 nouveau.blacklist=1 LANG=en_GB.UTF-8
```
```
aryasenna  ~  optirun glxinfo | grep NVIDIA
OpenGL vendor string: NVIDIA Corporation
OpenGL core profile version string: 4.5.0 NVIDIA 384.59
OpenGL core profile shading language version string: 4.50 NVIDIA
OpenGL version string: 4.5.0 NVIDIA 384.59
OpenGL shading language version string: 4.50 NVIDIA

```

```
 aryasenna ~  bumblebee-nvidia --check 


nvidia.ko compiled into in the kernel tree ok.
modinfo output for NVIDIA:

filename:       /lib/modules/4.12.5-300.fc26.x86_64/kernel/drivers/video/nvidia.ko
alias:          char-major-195-*
version:        384.59
supported:      external
license:        NVIDIA
srcversion:     31FF0349D3C7B1D9A62B474
alias:          pci:v000010DEd00000E00sv*sd*bc04sc80i00*
alias:          pci:v000010DEd*sv*sd*bc03sc02i00*
alias:          pci:v000010DEd*sv*sd*bc03sc00i00*
depends:        
vermagic:       4.12.5-300.fc26.x86_64 SMP mod_unload 
parm:           NVreg_Mobile:int
parm:           NVreg_ResmanDebugLevel:int
parm:           NVreg_RmLogonRC:int
parm:           NVreg_ModifyDeviceFiles:int
parm:           NVreg_DeviceFileUID:int
parm:           NVreg_DeviceFileGID:int
parm:           NVreg_DeviceFileMode:int
parm:           NVreg_UpdateMemoryTypes:int
parm:           NVreg_InitializeSystemMemoryAllocations:int
parm:           NVreg_UsePageAttributeTable:int
parm:           NVreg_MapRegistersEarly:int
parm:           NVreg_RegisterForACPIEvents:int
parm:           NVreg_CheckPCIConfigSpace:int
parm:           NVreg_EnablePCIeGen3:int
parm:           NVreg_EnableMSI:int
parm:           NVreg_TCEBypassMode:int
parm:           NVreg_UseThreadedInterrupts:int
parm:           NVreg_EnableStreamMemOPs:int
parm:           NVreg_MemoryPoolSize:int
parm:           NVreg_RegistryDwords:charp
parm:           NVreg_RegistryDwordsPerDevice:charp
parm:           NVreg_RmMsg:charp
parm:           NVreg_AssignGpus:charp

Check bbswitch kernel module...

bbswitch is loaded into the current kernel ok.

All checks completed successfully! NVIDIA driver appears to have compiled ok.

Documentation on bumblebee for RHEL / CentOS / fedora can be found at:
https://www.linux.ncsu.edu/bumblebee/


```

@arcivanov 
Reopened, what's the new issue name? just replace the kernel version in the title?

**Comment by arcivanov** [comment]
I have rebooted without `pcie_port_pm=off`, it does not work without it. That said, the `bumblebee-nvidia --check` succeeds.

```
bash-4.4$  bumblebee-nvidia --check 


nvidia.ko compiled into in the kernel tree ok.
modinfo output for NVIDIA:

filename:       /lib/modules/4.12.5-300.fc26.x86_64/kernel/drivers/video/nvidia.ko
alias:          char-major-195-*
version:        384.59
supported:      external
license:        NVIDIA
srcversion:     31FF0349D3C7B1D9A62B474
alias:          pci:v000010DEd00000E00sv*sd*bc04sc80i00*
alias:          pci:v000010DEd*sv*sd*bc03sc02i00*
alias:          pci:v000010DEd*sv*sd*bc03sc00i00*
depends:        
vermagic:       4.12.5-300.fc26.x86_64 SMP mod_unload 
parm:           NVreg_Mobile:int
parm:           NVreg_ResmanDebugLevel:int
parm:           NVreg_RmLogonRC:int
parm:           NVreg_ModifyDeviceFiles:int
parm:           NVreg_DeviceFileUID:int
parm:           NVreg_DeviceFileGID:int
parm:           NVreg_DeviceFileMode:int
parm:           NVreg_UpdateMemoryTypes:int
parm:           NVreg_InitializeSystemMemoryAllocations:int
parm:           NVreg_UsePageAttributeTable:int
parm:           NVreg_MapRegistersEarly:int
parm:           NVreg_RegisterForACPIEvents:int
parm:           NVreg_CheckPCIConfigSpace:int
parm:           NVreg_EnablePCIeGen3:int
parm:           NVreg_EnableMSI:int
parm:           NVreg_TCEBypassMode:int
parm:           NVreg_UseThreadedInterrupts:int
parm:           NVreg_EnableStreamMemOPs:int
parm:           NVreg_MemoryPoolSize:int
parm:           NVreg_RegistryDwords:charp
parm:           NVreg_RegistryDwordsPerDevice:charp
parm:           NVreg_RmMsg:charp
parm:           NVreg_AssignGpus:charp

Check bbswitch kernel module...

bbswitch is loaded into the current kernel ok.

All checks completed successfully! NVIDIA driver appears to have compiled ok.

Documentation on bumblebee for RHEL / CentOS / fedora can be found at:
https://www.linux.ncsu.edu/bumblebee/
```

**Comment by arcivanov** [comment]
@aryasenna are you running powertop?

**Comment by aryasenna** [comment]
@arcivanov OK at least it's working and indeed it's something to do with  `pcie_port_pm=off` 

Yes, I'm running powertop + tuned

**Comment by arcivanov** [comment]
@aryasenna what card do you have?

**Comment by aryasenna** [comment]
@arcivanov like I said in the first post: My video card is NVIDIA Quadro K1200M,

**Comment by arcivanov** [comment]
Sorry, missed it. Sigh. 

![](http://i0.kym-cdn.com/entries/icons/original/000/003/398/2013_6beb.jpeg)

**Comment by pkokoshnikov** [comment]
Hi guys, 
I'm faced with similar problem on fedora 27 kernel 4.13.9-300.fc27.x86_64
bumblebee-nvidia --check shows me that
```Warning! This NVIDIA driver has not compiled successfully before on kernel 4.13.9-300.fc27.x86_64!
